s4e-backend/src/main/java/pl/cyfronet/s4e/controller/AppUserController.java

```diff
@@ -65,13 +65,21 @@
     public void register(
             @RequestBody @Valid RegisterRequest registerRequest,
             HttpServletRequest request
-    ) throws RecaptchaException {
+    ) throws RecaptchaException, NotFoundException {
         val email = registerRequest.getEmail();
 
         validateRecaptcha(request);
 
         try {
             appUserService.register(registerRequest);
+
+            val token = registerRequest.getToken();
+            if (token != null) {
+                invitationService.confirmInvitation(email, token);
+                val confirmInvitationEvent = new OnConfirmInvitationEvent(token, LocaleContextHolder.getLocale());
+                eventPublisher.publishEvent(confirmInvitationEvent);
```

Comment:
I'm not a fan of breaking the SRP in this method, however I understand the workflow. Nevertheless, add a test which verifies, that if line 78 throws NFE, then creating a new user is rolled back